spec/states/destroy_card_spec.rb

```diff
@@ -0,0 +1,33 @@
+RSpec.describe States::DestroyCard do
+  let(:state) { described_class.new(context) }
+  let(:extant_account) { instance_double('Account', name: 'Andrii', login: 'andrii', password: '987654', age: '54') }
+  let(:context) { instance_double('Storage') }
+  let(:cards) { [instance_double('Universal', number: '4444111122225555', type: 'virtual')] }
+  let(:card_index) { 7 }
+  let(:wrong_index) { 357 }
+
+  describe '#destroy card' do
+    context 'true' do
+      before do
+        allow(extant_account).to receive(:card).and_return(cards)
+        allow(context).to receive(:extant_account).and_return(extant_account)
+      end
+
+      it 'when card valid' do
+        state.instance_variable_set(:@next_state, States::DestroyCard::MENU_STATE)
```

Comment:
I dont think that it is a right check. account_have_cards? is the main point in this code